### PR TITLE
Remove homegrown Mutable*

### PR DESCRIPTION
**Pull request by MishaDemianenko**

### CODE REVIEW

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/DijkstraBidirectional.java [hunk 1]
```diff
@@ -108,7 +109,8 @@ public DijkstraBidirectional( PathExpander expander, CostEvaluator<Double> costE
     private Traverser traverser( Node start, final Node end, PathInterest interest )
     {
         final MutableDouble shortestSoFar = new MutableDouble( Double.MAX_VALUE );
-        final MutableDouble startSideShortest = new MutableDouble( 0 );
+        final org.apache.commons.lang3.mutable.MutableDouble
+                startSideShortest = new org.apache.commons.lang3.mutable.MutableDouble( 0 );
```
Comment (burqen) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/DijkstraBidirectional.java]: Maybe this is a little bit to specific.

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/DijkstraBidirectional.java [hunk 2]
```diff
@@ -164,7 +166,7 @@ public TraversalMetadata metadata()
         private final boolean stopAfterLowestCost;
 
         DijkstraBidirectionalPathExpander( PathExpander source, MutableDouble shortestSoFar,
-                boolean stopAfterLowestCost, MutableDouble thisSideShortest, MutableDouble otherSideShortest,
+                boolean stopAfterLowestCost, org.apache.commons.lang3.mutable.MutableDouble thisSideShortest, MutableDouble otherSideShortest,
```
Comment (burqen) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/DijkstraBidirectional.java]: And this

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/util/DijkstraBranchCollisionDetector.java [hunk 3]
```diff
@@ -40,7 +41,7 @@
 
     public DijkstraBranchCollisionDetector( Evaluator evaluator,
             CostEvaluator costEvaluator,
-            MutableDouble shortestSoFar, double epsilon, Predicate<Path> pathPredicate )
+            org.apache.commons.lang3.mutable.MutableDouble shortestSoFar, double epsilon, Predicate<Path> pathPredicate )
```
Comment (burqen) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/util/DijkstraBranchCollisionDetector.java]: I'm super confident that we use `org.apache.commons.lang3.mutable.MutableDouble` now.

**Comment by MishaDemianenko** [comment]
@burqen cleaned up, sorry about that for some reason type migration left those...¯\_(ツ)_/¯

**Comment by burqen** [comment]
@MishaDemianenko Pushed some additional fixes.
Should be ready for merge now if you second those fixes.

**Pull request merged**